### PR TITLE
Block indices string

### PR DESCRIPTION
**Pull request by guidokanschat**
Add a string representation to `BlockIndices`.

Thus should be merged after #1083 and modified if the function gets renamed

### CODE REVIEW

File: include/deal.II/lac/block_indices.h [hunk 1]
```diff
@@ -423,6 +428,23 @@ BlockIndices::block_size (const unsigned int block) const
 
 
 inline
+std::string
+BlockIndices::str () const
+{
+  std::string result = "[" + std::to_string(n_blocks) + "->";
```
Comment (bangerth) [inline include/deal.II/lac/block_indices.h]: this won't compile with C++98. But you can use `Utilities::int_to_string()`.

**Comment by bangerth** [comment]
Indeed, same naming issue as in #1083.

**Comment by bangerth** [comment]
@guidokanschat : Can you change the name to accord with #1083?

**Comment by guidokanschat** [comment]
Renamed according to #1083 and new output added to test.

Should we deprecate the shift operator to LogStream?

**Comment by bangerth** [comment]
Your choice about `operator<<`. I think it doesn't hurt, but it probably also doesn't help very much.

**Pull request merged**